astroquery/mast/services.py

```diff
@@ -54,18 +54,19 @@ def _json_to_table(json_obj):
         if ignore_value == "NULL":
             ignore_value = None
         # making type adjustments
-        if col_type == "char" or col_type == "STRING":
+        if col_type == "char" or col_type == "STRING" or 'VARCHAR' in col_type or col_type == "NULL":
             col_type = "str"
             ignore_value = "" if (ignore_value is None) else ignore_value
         elif col_type == "boolean" or col_type == "BINARY":
             col_type = "bool"
         elif col_type == "unsignedByte":
             col_type = np.ubyte
-        elif col_type == "int" or col_type == "short" or col_type == "long" or col_type == "NUMBER":
+        elif (col_type == "int" or col_type == "short" or col_type == "long" or col_type == "NUMBER"
+                or col_type == 'INTEGER'):
             # int arrays do not admit Non/nan vals
             col_type = np.int64
             ignore_value = -999 if (ignore_value is None) else ignore_value
-        elif col_type == "double" or col_type == "float" or col_type == "DECIMAL":
+        elif col_type == "double" or col_type.lower() == "float" or col_type == "DECIMAL":
```

Comment:
I wonder if it would be safer to make all of our comparisons in this function case-insensitive.

Reply:
OK, I see the to lower has been moved to the top of the loop, so I am going to mark this conversation as resolved.